src/sqlfluff/rules/L066.py

```diff
@@ -0,0 +1,125 @@
+"""Implementation of Rule L066."""
+
+from typing import Optional
+
+from sqlfluff.core.rules.base import BaseRule, LintResult, RuleContext
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_groups,
+)
+
+
+@document_groups
+@document_configuration
+class Rule_L066(BaseRule):
+    """Enforce table alias lengths in from clauses and join conditions.
+
+    **Anti-pattern**
+
+    In this example, alias ``o`` is used for the orders table.
+
+    .. code-block:: sql
+
+        SELECT
+            SUM(o.amount) as order_amount,
+        FROM orders as o
+
+
+    **Best practice**
+
+    Avoid aliases. Avoid short aliases when aliases are necessary.
+
+    See also: L031.
+
+    .. code-block:: sql
+
+    SELECT
+        SUM(orders.amount) as order_amount,
+    FROM orders
+
+    SELECT
+        replacement_orders.amount,
+        previous_orders.amount
+    FROM
+        orders AS replacement_orders
+    JOIN
+        orders AS previous_orders
+        ON replacement_orders.id = previous_orders.replacement_id
+    """
+
+    groups = ("all",)
+    config_keywords = ["min_alias_length", "max_alias_length"]
+
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
+        """Identify aliases in from clause and join conditions.
+
+        Find base table, table expressions in join, and other expressions in select
+        clause and decide if it's needed to report them.
+        """
+
+        self.min_alias_length: Optional[int]
+        self.max_alias_length: Optional[int]
+
+        if context.segment.is_type("select_statement"):
+            children = context.functional.segment.children()
+
+            from_expression_elements = children.recursive_crawl(
+                "from_expression_element"
+            )
```

Comment:
L011 looks at both `from_expression_element` and `merge_statement`. Should we consider that here?

If so, should we move these to `_target_elems` class variable?

Reply:
Let's leave this for now and merge as is and come back to this if it's needed.